src/sagemaker/serve/builder/jumpstart_builder.py

```diff
@@ -431,25 +431,47 @@ def tune_for_tgi_jumpstart(self, max_tuning_duration: int = 1800):
             sharded_supported=sharded_supported, max_tuning_duration=max_tuning_duration
         )
 
+    def set_deployment_config(self, config_name: Optional[str]) -> None:
+        """Sets the deployment config to apply to the model.
+
+        Args:
+            config_name (Optional[str]):
+                The name of the deployment config. Set to None to unset
+                any existing config that is applied to the model.
+        """
+        self.pysdk_model.set_deployment_config(config_name)
+
+    def get_deployment_config(self) -> Optional[Dict[str, Any]]:
+        """Gets the deployment config to apply to the model.
+
+        Returns:
+            Union[Dict[str, Any], None]: Deployment config to apply to this model.
+        """
+        return getattr(self, "pysdk_model", self._create_pre_trained_js_model()).deployment_config
+
     def display_benchmark_metrics(self):
         """Display Markdown Benchmark Metrics for deployment configs."""
-        self.pysdk_model.display_benchmark_metrics()
+        getattr(
+            self, "pysdk_model", self._create_pre_trained_js_model()
+        ).display_benchmark_metrics()
```

Comment:
why do we need to use `getattr` here, and this is guaranteed from the constructor here https://github.com/aws/sagemaker-python-sdk/blob/faba3c825f6994f04762031c0f12361379fe7627/src/sagemaker/serve/builder/jumpstart_builder.py#L88.

and the JSModel instance needs to be assigned to `self.pysdk_model`

Reply:
I think this has something to do with the method resolution order in python, especially when we have such a complex class structure. IIRC, if not specified, it will only invoke the init method from the first parent class. And that's actually the case when I tested without this fix, ie.
```
model_builder = ModelBuilder(
    model=model_id,
    schema_builder=my_schema,
)
configs = model_builder.list_deployment_configs()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[5], line 1
----> 1 configs = model_builder.list_deployment_configs()

File ~/anaconda3/envs/python3/lib/python3.10/site-packages/sagemaker/serve/builder/jumpstart_builder.py:444, in JumpStart.list_deployment_configs(self)
    438 def list_deployment_configs(self) -> List[Dict[str, Any]]:
    439     """List deployment configs for ``This`` model in the current region.
    440 
    441     Returns:
    442         List[Dict[str, Any]]: A list of deployment configs.
    443     """
--> 444     return self.pysdk_model.list_deployment_configs()

AttributeError: 'ModelBuilder' object has no attribute 'pysdk_model'
```